plugins/inputs/prometheus/prometheus.go

```diff
@@ -437,9 +438,26 @@ func (p *Prometheus) gatherURL(u URLAndAddress, acc telegraf.Accumulator) error
 		return fmt.Errorf("%q returned HTTP status %q", u.URL, resp.Status)
 	}
 
-	body, err := io.ReadAll(resp.Body)
-	if err != nil {
-		return fmt.Errorf("error reading body: %w", err)
+	var body []byte
+	if p.ContentLengthLimit != 0 {
+		// To determine whether io.ReadAll() ended due to EOF or reached the specified limit,
+		// read up to the specified limit plus one extra byte, and then make a decision based
+		// on the length of the result.
+		lr := io.LimitReader(resp.Body, p.ContentLengthLimit+1)
+
+		body, err = io.ReadAll(lr)
+		if err != nil {
+			return fmt.Errorf("error reading body: %w", err)
+		}
+		if int64(len(body)) > p.ContentLengthLimit {
```

Comment:
Maybe have `ContentLengthLimit` to be just `int`, so this can be:
```suggestion
		if len(body) > p.ContentLengthLimit {
```

Reply:
I've changed the type for ContentLengthLimit to `config.Size` which is a Defined Type of int64. Therefore, I think there is no need to make any changes regarding this type conversion.